scarlet/frame.py

```diff
@@ -180,99 +185,74 @@ def from_observations(observations, target_psf = None, target_wcs = None, obs_id
-                observations[c] = obs.make_LowRes()
-            # Limits that include all observations relative to target_wcs
-            obs_coord = resampling.get_to_common_frame(obs, target_wcs)
-            if int(np.min(obs_coord[0])) < y_min:
+            if (obs is not obs_ref) and (type(obs) is not 'LowResObservation'):
+                observations[c] = obs.get_LowRes()
+                # Limits that include all observations relative to target_wcs
+                obs_coord = resampling.get_to_common_frame(obs, target_wcs)
                 y_min = np.min(obs_coord[0])
-            if int(np.min(obs_coord[1])) < x_min:
                 x_min = np.min(obs_coord[1])
-            #The upper limit is the bottom right corner
-            if int(np.max(obs_coord[0])) > y_max:
                 y_max = np.max(obs_coord[0])
-            if int(np.max(obs_coord[1])) > x_max:
                 x_max = np.max(obs_coord[1])
-
-        # Margin in pixels
-        fat_pixel_size = (fat_psf_size/h).astype(int)
-        #Padding by the size of the psf
-        if fat_pixel_size % 2 != 0:
-            fat_pixel_size += 1
-
-        offset = np.array([y_min, x_min])
-        # Shape of the box that encompasses all observations
-        ny = (y_max - offset[0] + 1).astype(int)
-        nx = (x_max - offset[1] + 1).astype(int)
-
-        coord_frame = (np.indices((ny,nx)))
-        footprint = 0
-        for obs in observations:
-            if obs.frame.wcs != target_wcs:
-                coord_2obs = np.around(resampling.convert_coordinates(coord_frame, target_wcs, obs.frame.wcs))
-            else:
-                coord_2obs = coord_frame
-            footprint += (coord_2obs[0] >= -np.finfo(float).eps) * (coord_2obs[1] >= -np.finfo(float).eps) *\
-                (coord_2obs[0] <= obs.frame.shape[-2]) * (coord_2obs[1] <= obs.frame.shape[-1])
-
-        if coverage is 'union':
-            coord_cover = np.where(footprint != 0)
-
-        elif coverage is 'intersection':
-            coord_cover = (np.where(footprint == np.max(footprint)))
-        #Corners of the footprint with padding by the psf size
-        y_fmin = np.min(coord_cover[0]) - fat_pixel_size / 2 + offset[0]
-        x_fmin = np.min(coord_cover[1]) - fat_pixel_size / 2 + offset[1]
-        y_fmax = np.max(coord_cover[0]) + fat_pixel_size / 2 + offset[0]
-        x_fmax = np.max(coord_cover[1]) + fat_pixel_size / 2 + offset[1]
-        # Shape of the resulting frame
-        ny = (y_fmax - y_fmin + 1).astype(int)
-        nx = (x_fmax - x_fmin + 1).astype(int)
-        frame_shape =(len(channels), ny, nx)
-
-        fbox = Box(frame_shape, origin = (0,np.around(y_fmin).astype(int), np.around(x_fmin).astype(int)))
+                if coverage == 'union':
+                    ref_box = ref_box | Box((obs.frame.C, y_max - y_min + 1, x_max - x_min + 1),
+                                            origin = (0, y_min, x_min))
+                elif coverage == 'union':
```

Comment:
`if coverage == 'union'` and `elif coverage == 'union'` ensure that the following block will never run

Reply:
You can also probably simplify this to something like
```
new_box = Box((obs.frame.C, y_max - y_min + 1, x_max - x_min + 1), origin = (0, y_min, x_min))
if coverage == 'union':
    ref_box |= new_box
else:
    ref_box &= new_box
```

since you already checked that `coverage` is either `union` or `intersection`.

Reply:
Good catch, that's a major mistake!
And thanks for the simplification! That's that kind of stuff that I should have caught by doing a good second pass.